src/irc/IrcServer.ts

```diff
@@ -770,7 +772,10 @@ export class IrcServer {
         }
 
         this.addresses = config.additionalAddresses || [];
-        this.addresses.push(this.domain);
+        // Don't include the original domain if not configured to.
+        if (!config.onlyAdditionalAddresses) {
+            this.addresses.push(this.domain);
+        }
```

Comment:
If you don't want runtime errors (division by zero), you need to throw here if `.length === 0`.